stablehlo/reference/Ops.cpp

```diff
@@ -68,6 +81,44 @@ Tensor eval(FloorOp op, const Tensor &operand) {
   return result;
 }
 
+Tensor eval(IotaOp op) {
+  Tensor result(op.getType());
+  Type elType = result.getType().getElementType();
+  if (!isSupportedIntegerType(elType) && !isSupportedFloatType(elType) &&
+      !isSupportedComplexType(elType))
+    report_fatal_error(invalidArgument("Unsupported element type: %s",
+                                       debugString(elType).c_str()));
+  uint64_t iotaDimension = op.getIotaDimension();
+  llvm::RoundingMode rm = APFloat::rmNearestTiesToEven;
+  bool roundingErr;
+  for (auto it = result.index_begin(); it != result.index_end(); ++it) {
+    auto iota = (*it)[iotaDimension];
+    if (isSupportedIntegerType(elType)) {
+      result.set(*it, Element(elType, APInt(elType.getIntOrFloatBitWidth(),
```

Comment:
Is it fine to consider all int types as unsigned ints here? 
I agree that these values are handled property, in `Tensor::get and tensor::set`, with or w/o sign extension based on the type being signed or unsigned, but I prefer generating the `APints` here with proper signed. First it would be agnostic of how we handle these values later and brings the implementation sync with the spec. 

Wdyt?

Reply:
I'll create separate cases for signed and unsigned integers as that would make it more aligned with the spec's notion of implementation-defined for signed and unsigned integer element types.